File: Unit_tests_python/EX-04-Calculate_stats/Copilot.py
import math

def calculate_stats(list1, list2, list3):
    stats = {}
    
    # Calculate mean
    mean1 = sum(list1) / len(list1)
    mean2 = sum(list2) / len(list2)
    mean3 = sum(list3) / len(list3)
    
    # Calculate standard deviation
    stddev1 = math.sqrt(sum((x - mean1) ** 2 for x in list1) / len(list1))
    stddev2 = math.sqrt(sum((x - mean2) ** 2 for x in list2) / len(list2))
    stddev3 = math.sqrt(sum((x - mean3) ** 2 for x in list3) / len(list3))
    
    # Add mean and standard deviation to dictionary
    stats['mean'] = [mean1, mean2, mean3]
    stats['stddev'] = [stddev1, stddev2, stddev3]
    
    return stats

File: Unit_tests_lua/EX-04-Calculate_stats/Copilot.lua
-- Import the necessary modules
local math = require("math")

-- Function to calculate mean
local function calculate_mean(list)
    local sum = 0
    for _, value in ipairs(list) do
        sum = sum + value
    end
    return sum / #list
end

-- Function to calculate standard deviation
local function calculate_stddev(list, mean)
    local sum = 0
    for _, value in ipairs(list) do
        sum = sum + (value - mean) ^ 2
    end
    return math.sqrt(sum / #list)
end

-- Main function to calculate stats
local function calculate_stats(list1, list2, list3)
    local stats = {}
    stats.mean = {
        calculate_mean(list1),
        calculate_mean(list2),
        calculate_mean(list3)
    }
    stats.stddev = {
        calculate_stddev(list1, stats.mean[1]),
        calculate_stddev(list2, stats.mean[2]),
        calculate_stddev(list3, stats.mean[3])
    }
    return stats
end
    
return calculate_stats